kimchi/src/circuits/gate.rs

```diff
@@ -68,40 +69,67 @@ pub struct SingleLookup<F> {
 /// analogously using `joint_combiner`.
 ///
 /// This function computes that combined value.
-pub fn combine_table_entry<'a, F: Field, I: DoubleEndedIterator<Item = &'a F>>(
+pub fn combine_table_entry<
+    'a,
+    F: 'a + Zero + One + Add + Mul + Clone,
+    I: DoubleEndedIterator<Item = &'a F>,
+>(
     joint_combiner: F,
     v: I,
 ) -> F {
-    v.rev().fold(F::zero(), |acc, x| joint_combiner * acc + x)
+    v.rev()
+        .fold(F::zero(), |acc, x| joint_combiner.clone() * acc + x.clone())
```

Comment:
if you add Copy instead of Clone you can remove the clone here (Field is Copy)

Reply:
`F` is copy but `Expr` isn't (and we use `JointCombiner` rather than a field element there).

Reply:
ah I see. My tolerance for generics is usually pretty low ^^ I'm wondering how we can make this clearer. Maybe by changing `F` with `F_or_Expr` or something. Food for thought

Reply:
and then creating an actual trait `F_or_Expr : Zero + One + ...` and implement it on what we need